### ISSUE TITLE
Ranker not utilizing max flow appropriately?

### ISSUE THREAD

**Issue by amykglen** [RTXteam/RTX#1630]
discovered this while working on #1627, but writing it up here with a simpler example

for our classic acetaminophen example query: https://arax.ncats.io/?r=23292

you can see that the first result (score of 0.862) has only one edge, with one publication:

<img width="1194" alt="Screen Shot 2021-08-26 at 8 32 14 PM" src="https://user-images.githubusercontent.com/49423686/131067717-f4c22b24-6452-4b78-91b9-515191635efb.png">

but result 30 (also a score of 0.862) has many edges, most of which have 1 or more publications:

<img width="1200" alt="Screen Shot 2021-08-26 at 8 33 23 PM" src="https://user-images.githubusercontent.com/49423686/131067773-576f1a9c-f702-47e8-979c-90f404bd01c8.png">

why does result 30 not have a higher score than result 1? I thought that the ranker considers max flow, which should differentiate these results.. the fact that they have the same exact score makes it seem like max flow isn't being considered?

**Comment by amykglen** [comment]
So I think I found the problem here: when a result has multiple parallel edges fulfilling a given qedge, the `_get_weighted_graph_networkx_from_result_graph()` method was essentially throwing out all but one of those parallel edges. i.e., the 'weight' being assigned to that edge in the weighted result graph was the confidence of only _one_ of the parallel edges (an essentially randomly selected one), when really I think the weight should be the _sum_ of the confidences of the parallel edges.

I pushed a fix in 0fa34f4. Anyone with more Ranker expertise should feel free to jump in if something sounds off here, but I confirmed this resolves the problem reported in this issue: the top 4 results for the query are now 'CYP2E1', 'PTGS1', 'PTGS2', and 'CYP1A2' (which seem much improved).
